lib/AST/ASTPrinter.cpp

```diff
@@ -2011,6 +2026,27 @@ void PrintAST::visitExtensionDecl(ExtensionDecl *decl) {
     printExtension(decl);
 }
 
+/// Determines if we should print the text of the initializer expression.
+/// There's a very narrow case when we would:
+/// If the decl:
+///   - has an initializer expression
+///   - is @usableFromInline
+/// If the parent type:
+///   - is @_fixed_layout (or effectively fixed-layout)
```

Comment:
What does effectively fixed-layout mean? Only types that have an explicit @_fixed_layout attribute need this treatment. If the module is not resilient and the type is implicitly @_fixed_layout, you cannot print the initial value because it might reference private things. However in that case the initializer symbol is public, so you don't need to emit it yourself.

Reply:
If the type is explicitly fixed_layout and the property is non-public but usable from inline, should it still print the value?

Reply:
Yes, and if the property is public too. @usableFromInline makes things public for ABI purposes but not public in the language itself. So “@usableFromInline but not public” is not a useful category of declaration. It’s always “public” or “public or @usableFromInline”.